R/layer_add_forecast_date.R

```diff
@@ -93,23 +93,24 @@ slather.layer_add_forecast_date <- function(object, components, workflow, new_da
     )
     object$forecast_date <- max_time_value
   }
-  as_of_pre <- attributes(workflows::extract_preprocessor(workflow)$template)$metadata$as_of
-  as_of_fit <- workflow$fit$meta$as_of
-  as_of_post <- attributes(new_data)$metadata$as_of
 
-  as_of_date <- as.Date(max(as_of_pre, as_of_fit, as_of_post))
+  expected_time_type <- attr(
+    workflows::extract_preprocessor(workflow)$template, "metadata"
+  )$time_type
+  if (expected_time_type == "week") expected_time_type <- "day"
+  check <- validate_date(object$forecast_date, expected_time_type)
 
-  if (object$forecast_date < as_of_date) {
```

Comment:
This is just a warning. One that almost always triggers in examples (because the `as_of` is whenever we downloaded the example data). I think it should go away.